src/ai/defaultai.cc

```diff
@@ -2926,6 +2977,17 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					number_of_same_nearby += same_it->second;
 				}
 
+				// Considering distance to a warehouse NOCOM
+				if (!bo.requires_supporters && !bo.inputs.empty()) {
+					prio -= wh_distance_malus;
+
+				}
+				// some "independent" space consumers can be pushed to bigger distance from wh
+				if (bo.is(BuildingAttribute::kSpaceConsumer) && bo.inputs.empty() &&
+				    bo.is(BuildingAttribute::kSpaceConsumer)) {
```

Comment:
1. the space consumer entry is doubled.
2. I prefer not doing this in general but specific for each subtype

Reply:
The second one was a mistake, I changed it to
!bo.is(BuildingAttribute::kSupportingProducer
But as I said, I am not very happy about being very specific about this malus